### PR TITLE
Allow viable/strict promotion even if periodic or docker-release-builds jobs are failing

### PR DESCRIPTION
**Pull request by ZainRizvi**
Allow `viable/strict` promotion even if `periodic` or `docker-release-builds` jobs are failing

**Why?** Those jobs only run occasionally and for all we know the current viable/strict commit may already include the errors that the above cron based workflows may have later detected.  Blocking the viable/strict upgrade because of these scheduled jobs doesn't really offer any value, it just leads to people getting older PRs when they try to fork off of viable/strict without guaranteeing an improvement in test quality

Though frankly, the current situation is worse than that.

Assume the branch history looks like A -> B 

A is the current `viable/strict` commit
B is a commit that failed some `periodic` test, so `viable/strict` wasn't upgraded to B

Now lets say there's a commit C that gets merged. C neither contains a fix for the failing periodic build, nor does a scheduled periodic workflow run against C. The branch becomes A -> B -> C

In the above scenario, today we will promote `viable/strict` to C since there was no failing workflow there!!! Even though it didn't actually fix what was broken with B!

In short, avoiding the upgrade to B really doesn't make any sense today and we shouldn't do it.

### CODE REVIEW

**Review by janeyx99 — APPROVED** [review]
seems fine
